base/applications/osk/settings.c

```diff
@@ -37,133 +31,48 @@ BOOL LoadDataFromRegistry(VOID)
-
-    /* Query the key */
-    lResult = RegQueryValueExW(hKey,
-                               L"IsEnhancedKeyboard",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
-
-    /* Load the dialog layout value */
-    Globals.bIsEnhancedKeyboard = (dwValue != 0);
-
-    /* Query the key */
-    lResult = RegQueryValueExW(hKey,
-                               L"ClickSound",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
-
-    /* Load the sound on click value event */
-    Globals.bSoundClick = (dwValue != 0);
-
-    /* Query the key */
-    lResult = RegQueryValueExW(hKey,
-                               L"WindowLeft",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
-    }
-
-    /* Load the X value data of the dialog's coordinate */
-    Globals.PosX = dwValue;
-
-    lResult = RegQueryValueExW(hKey,
-                               L"WindowTop",
-                               0,
-                               0,
-                               (BYTE *)&dwValue,
-                               &cbData);
-
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
-    {
-        /* Bail out and return FALSE if we fail */
-        RegCloseKey(hKey);
-        return FALSE;
+        /* Bail out */
+        DPRINT("LoadDataFromRegistry(): Failed to open the application's key! (Error - %li)\n", lResult);
+        return lResult;
     }
 
-    /* Load the Y value data of the dialog's coordinate */
-    Globals.PosY = dwValue;
-
+    /* Load the specific value based on the parameter caller, lpValueDataName */
     lResult = RegQueryValueExW(hKey,
-                               L"AlwaysOnTop",
+                               lpValueDataName,
                                0,
                                0,
                                (BYTE *)&dwValue,
                                &cbData);
 
-    if (lResult != ERROR_SUCCESS || cbData != sizeof(dwValue))
+    if (lResult != ERROR_SUCCESS)
     {
-        /* Bail out and return FALSE if we fail */
+        /* Is the buffer's size too small to query the required data? */
+        if (cbData != sizeof(dwValue))
```

Comment:
By moving this inside here you turned the above `||` into an `&&`. You want it after the `lResult != ERROR_SUCCESS` block instead.